Logcollector scanning IT: The ValueError was raised when attempting to stop the 'wazuh-logcollector' daemon

### Description

It has been detected in https://github.com/wazuh/wazuh/issues/20479 where Testing logcollector integration in Windows Agent, the following connection error were found:

```
>           raise ValueError(f"Error when executing {action} in daemon {daemon}. Exit status: {result}")
E           ValueError: Error when executing stop in daemon wazuh-logcollector. Exit status: 2

C:\Python311\Lib\site-packages\wazuh_testing\tools\services.py:147: ValueError

```

### Evidences

- **Build**: https://ci.wazuh.info/job/Test_integration/45056/
- **Result**: https://s3.amazonaws.com/ci.wazuh.com/data/Test_integration/B45044/agent_windows_html_report_Test_integration_B45044.zip

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@ All notable changes to this project will be documented in this file.
 
 ### Changed
 
+- Enhance control_service error handling on windows agents. ([#4733](https://github.com/wazuh/wazuh-qa/pull/4733)) \- (Framework)
 - Add XFAIL mark to Cluster reliability logs test. ([#4706](https://github.com/wazuh/wazuh-qa/pull/4706)) \- (Tests)
 
 ## [4.7.0] - TBD

--- a/deps/wazuh_testing/wazuh_testing/tools/services.py
+++ b/deps/wazuh_testing/wazuh_testing/tools/services.py
@@ -89,14 +89,18 @@ def control_service(action, daemon=None, debug_mode=False):
                 command = subprocess.run(["net", action, "WazuhSvc"], stderr=subprocess.PIPE)
                 result = command.returncode
                 if result != 0:
-                    if action == 'stop' and 'The Wazuh service is not started.' in command.stderr.decode():
+                    error = command.stderr.decode()
+                    if 'The service is starting or stopping' in error:
+                        time.sleep(1)
+                        continue
+                    if action == 'stop' and 'The Wazuh service is not started.' in error:
                         result = 0
                         break
-                    if action == 'start' and 'The requested service has already been started.' \
-                       in command.stderr.decode():
+                    if action == 'start' and 'The requested service has already been started.' in error:
                         result = 0
                         break
-                    elif "System error 109 has occurred" not in command.stderr.decode():
+                    elif "System error 109 has occurred" not in error:
+                        print(f"Unexpected error when control_service failed with the following error: {error}")
                         break
     else:  # Default Unix
         if daemon is None:

